core/src/engine/io/tls_wrapper.cpp

```diff
@@ -540,6 +541,18 @@ TlsWrapper TlsWrapper::StartTlsServer(
     LOG_INFO() << "Client SSL cert will not be verified";
   }
 
+  if(!certificate_chain_path.empty()) {
+      if(1 !=SSL_CTX_use_certificate_chain_file(ssl_ctx.get(), certificate_chain_path.c_str()))
+	  {
+          throw TlsException(crypto::FormatSslError("Failed to set up server TLS wrapper: SSL_CTX_use_certificate_chain_file"));
+      }
+  }
+  else {
+      if (1 != SSL_CTX_use_certificate(ssl_ctx.get(), cert.GetNative())) {
```

Comment:
Thos block is duplicated down below. Looks like the latter is redundant

Reply:
removed